### PR TITLE
Visibility query constructor for workflow shadower

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
Visibility query constructor for workflow shadower

**Why?**
Visibility query constructor for workflow shadower

**How did you test it?**
Tested locally

**Potential risks**
N/A

### CODE REVIEW

File: internal/query_constructor.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	q.appendPartialQuery(strings.Join(workflowStatusQueries, " or "))
+	return q
+}
+
+func (q *queryConstructorImpl) StartTime(minStartTime, maxStartTime time.Time) QueryConstructor {
+	startTimeQueries := make([]string, 0, 2)
+	if !minStartTime.IsZero() {
+		startTimeQueries = append(startTimeQueries, fmt.Sprintf(keyStartTime+` >= %v`, minStartTime.UnixNano()))
+	}
+	if !maxStartTime.Equal(maxTimestamp) {
```
Comment (yux0) [inline internal/query_constructor.go:101]: What if the maxStartTime is zero?

File: internal/query_constructor.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		startTimeQueries = append(startTimeQueries, fmt.Sprintf(keyStartTime+` <= %v`, maxStartTime.UnixNano()))
+	}
+
+	q.appendPartialQuery(strings.Join(startTimeQueries, " and "))
+	return q
+}
+
+func (q *queryConstructorImpl) Query() string {
```
Comment (yux0) [inline internal/query_constructor.go:109]: maybe named this query builder and build() function?

File: internal/query_constructor.go (hunk 1 above)
Comment (yycptt) [reply internal/query_constructor.go:101]: This component assumes whatever value got passed in should be used in the final query. Shadower is responsible for converting and filling unspecified values such as sampling rate, max timestamp etc. but that logic is not in this diff.

File: internal/query_constructor.go (hunk 2 above)
Comment (yycptt) [reply internal/query_constructor.go:109]: Makes sense. Will update.

Comment (yycptt) [reply internal/query_constructor.go]: Renamed to `QueryBuilder` and `Build()`

**Review by yux0 — APPROVED** [review]

**Pull request merged**